File: src/net/GOOM.TR.MyRetail.NET/GOOM.TR.MyRetail.NET/Controllers/ProductsController.cs
using GOOM.TR.MyRetail.NET.Models;
using GOOM.TR.MyRetail.NET.Services;
using Microsoft.AspNetCore.Mvc;
using System.Threading.Tasks;

namespace GOOM.TR.MyRetail.NET.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class ProductsController : ControllerBase
    {
        private readonly IProductService productService;

        public ProductsController(IProductService productService)
        {
            this.productService = productService;
        }

        [HttpGet("{id}")]
        public async Task<Product> Get([FromRoute] long id)
        {
            return await productService.GetProductAsync(id);
        }

        [HttpPut("{id}/current_price")]
        public async Task<Price> SetCurrentPrice([FromRoute] long id, [FromBody] Price currentPrice)
        {
            return await productService.SetProductCurrentPriceAsync(id, currentPrice);
        }

        [HttpDelete("{id}/current_price")]
        public async Task<bool> DeleteCurrentPrice([FromRoute] long id)
        {
            return await productService.DeleteProductCurrentPriceAsync(id);
        }
    }
}

File: src/java/src/main/java/com/goom/tr/myretail/controllers/ProductsController.java
package com.goom.tr.myretail.controllers;

import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.goom.tr.myretail.models.*;
import com.goom.tr.myretail.services.IProductService;

@RestController
@RequestMapping("/products")
public class ProductsController {
    private IProductService productService;

    public ProductsController(IProductService productService)
    {
        this.productService = productService;
    }

    @GetMapping("{id}")
    public Product Get(@PathVariable long id)
    {
        return productService.GetProduct(id);
    }

    @PutMapping("{id}/current_price")
    public Price SetCurrentPrice(@PathVariable long id, @RequestBody Price currentPrice)
    {
        return productService.SetProductCurrentPrice(id, currentPrice);
    }

    @DeleteMapping("{id}/current_price")
    public boolean DeleteCurrentPrice(@PathVariable long id)
    {
        return productService.DeleteProductCurrentPrice(id);
    }
}
